app/workers/web_mentions/send_web_mention.rb

```diff
@@ -0,0 +1,17 @@
+module WebMentions
+  class SendWebMention
+    include Sidekiq::Worker
+    sidekiq_options queue: :low_priority, retry: 15, lock: :until_executing
+
+    def perform(comment_id)
+      commentable = Comment.find_by(id: comment_id)&.commentable
+      return unless commentable
+
+      return unless commentable.support_webmentions
```

Comment:
That sounds good. Maybe then we could also remove ` WebMentions::CheckWebMentionSupport`, since that calls the same handler to set the DB column. If we do that, we probably also don't need the validation on `Article`, so the simplified flow would: in `after_commit` of a commit, enqueue this job (which we already do unconditionally anyway), then just rely on this and the handler to do the right thing:

```ruby
    def perform(comment_id)
      commentable = Comment.find_by(id: comment_id)&.commentable
      return unless commentable
      WebMentions::WebMentionHandler.call(commentable)
    end
```

This way `Article` itself doesn't need to know about web mentions at all and the full logic (early return when there's no canonical URL, building a full article URL with the `URL` module, etc) is in one place.

Reply:
Small suggestion if we decide to keep it: `supports_webmentions` instead of `support_webmentions`